modules/internal/ChapelArray.chpl

```diff
@@ -3163,36 +3147,36 @@ module ChapelArray {
     //return help(1);
   }
 
-  proc _getRankChangeRanges(args) {
-    proc _tupleize(x) {
-      var y: 1*x.type;
-      y(1) = x;
-      return y;
-    }
+  proc _getRankChangeRanges(args, downdom) {
+    return collectRanges(1);
+
     proc collectRanges(param dim: int) {
```

Comment:
See my note around 1137. I think collectRanges is unnecessarily complicated and could be much simpler in a for-loop form and in a form that separated the matter of intersecting from the matter of leaving out the collapsed dimensions. Moreover, since this is only ever called from the domain rank-change `this`, it seems odd to have it all the way down here. Could it be a utility function in ArrayViewRankChange.chpl ? Or an inner function in the `this` method? Or just a series of loops that is easier to follow?